diff --git a/Pipfile b/Pipfile
new file mode 100644
index 0000000000000000000000000000000000000000..5f5e774326331f2fe303ff2669c6461b5aefd7c5
--- /dev/null
+++ b/Pipfile
@@ -0,0 +1,25 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+aiohttp = "~=3.8"
+click = "~=8.1.3"
+easyocr = "~=1.7"
+ecdsa = "~=0.18"
+littlefs-python = "==0.8"
+numpy = "==1.26.4"
+pillow = "~=10.4"
+polib = "~=1.2"
+pre-commit = "*"
+pytest-asyncio = "~=0.21"
+pytest = "~=7.3.2"
+pyyaml = "~=6.0"
+qoi = "~=0.5"
+requests = "==2.32.3"
+
+[dev-packages]
+
+[requires]
+python_version = "3.10"
