diff --git a/src/guiapi/src/window_msgbox.cpp b/src/guiapi/src/window_msgbox.cpp
index e94bdaa08f2e15dd3607d906783f80e51df4cfaf..02bbfbe275a78eb61db37147a2d065833a82ed0f 100644
--- a/src/guiapi/src/window_msgbox.cpp
+++ b/src/guiapi/src/window_msgbox.cpp
@@ -233,9 +233,9 @@ MsgBoxIconnedError::MsgBoxIconnedError(Rect16 rect, const PhaseResponses &resp,
     text.SetAlignment(Align_t::LeftCenter());
     icon.SetRect(getIconRect());
     AdjustLayout(text, icon);
-    SetBackColor(COLOR_ORANGE);
-    text.SetBackColor(COLOR_ORANGE);
-    icon.SetBackColor(COLOR_ORANGE);
+    SetBackColor(COLOR_PRUSA_ORANGE);
+    text.SetBackColor(COLOR_PRUSA_ORANGE);
+    icon.SetBackColor(COLOR_PRUSA_ORANGE);
 
     if (!pButtons) { // pButtons can never be null
         assert("unassigned msgbox");
