diff --git a/src/common/marlin_client.cpp b/src/common/marlin_client.cpp
index 6efd24411949e9daa41259646aa40104181353f2..44fa8f56611080e9a0c8bf58528706fa41511a50 100644
--- a/src/common/marlin_client.cpp
+++ b/src/common/marlin_client.cpp
@@ -327,6 +327,7 @@ bool is_print_started() {
         case State::Aborted:
         case State::Finished:
             // Went to idle - refused by GUI
+            // TODO need to close the screen somehow?
             return false;
         default:
             // Doing something else ‒ there's a lot of states where we are printing.
