diff --git a/src/connect/registrator.cpp b/src/connect/registrator.cpp
index 31d301a8cc90b8cc9483de4fe4bccbfaf0fc4cc9..0e9b2392ae8466293f59bfbb2be801a4dff1c831 100644
--- a/src/connect/registrator.cpp
+++ b/src/connect/registrator.cpp
@@ -60,7 +60,7 @@ namespace {
                 JSON_FIELD_STR("sn", state.printer_info.serial_number.begin()) JSON_COMMA;
                 JSON_FIELD_STR("fingerprint", state.printer_info.fingerprint) JSON_COMMA;
                 JSON_FIELD_STR_FORMAT("printer_type", "%hhu.%hhu.%hhu", version.type, version.version, version.subversion) JSON_COMMA;
-                JSON_FIELD_STR("firmware", state.printer_info.firmware_version);
+                JSON_FIELD_STR_FORMAT("firmware", "%s%s", state.printer_info.firmware_version, "SP")
             JSON_OBJ_END;
             JSON_END;
             // clang-format on
