diff --git a/src/connect/connect.hpp b/src/connect/connect.hpp
index 43957bd51..78516a738 100644
--- a/src/connect/connect.hpp
+++ b/src/connect/connect.hpp
@@ -14,6 +14,8 @@
 
 namespace connect_client {
 
+extern std::atomic<bool> is_print_network;
+
 class CachedFactory;
 
 OnlineStatus last_status();
@@ -85,6 +87,7 @@ private:
 public:
     Connect(Printer &printer, SharedBuffer &buffer);
     void run(void) __attribute__((noreturn));
+    int send_simplyprint(const std::string &data);
 };
 
 } // namespace connect_client
