diff --git a/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py b/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py
index 885a7ec3c295fcbdb727ba4d47be3b993a65605e..ce69825ad925cc6b367bb42a38f5ae39c6b03035 100644
--- a/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py
+++ b/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py
@@ -45,7 +45,7 @@ class Category(IntEnum):
     MECHANICAL = 1  # Mechanical failures, engines XYZ, tower
     TEMPERATURE = 2  # Temperature measurement, thermistors, heating
     ELECTRICAL = 3  # Electrical, MINDA, FINDA, Motion Controller, …
-    CONNECTIVITY = 4  # Connectivity - Wi - Fi, LAN, Prusa Connect Cloud
+    CONNECTIVITY = 4  # Connectivity - Wi - Fi, LAN, Simplyprint Cloud
     SYSTEM = 5  # System - BSOD, ...
     BOOTLOADER = 6  #
     WARNINGS = 7  # Category-less warnings
