diff --git a/src/gui/MItem_menus.cpp b/src/gui/MItem_menus.cpp
index d5e9ca33d..b51782389 100644
--- a/src/gui/MItem_menus.cpp
+++ b/src/gui/MItem_menus.cpp
@@ -86,6 +86,8 @@
 #include <config_store/store_instance.hpp>
 
 /*****************************************************************************/
+//TODO add code to handle setting the lock screen password
+
 // MI_VERSION_INFO
 MI_VERSION_INFO::MI_VERSION_INFO()
     : IWindowMenuItem(_(label), nullptr, is_enabled_t::yes, is_hidden_t::no, expands_t::yes) {
