diff --git a/src/gui/ScreenPrintingModel.hpp b/src/gui/ScreenPrintingModel.hpp
index 52d667d6f..0769a1fef 100644
--- a/src/gui/ScreenPrintingModel.hpp
+++ b/src/gui/ScreenPrintingModel.hpp
@@ -28,6 +28,9 @@ protected:
         Home,
         Reprint,
         Disconnect,
+        Print_OK,
+        Print_Failed,
+        Bed_Clear,
         _count,
         _last = _count - 1,
     };
@@ -44,6 +47,9 @@ protected:
         Home,
         Skip,
         Disconnect,
+        Print_OK,
+        Print_Failed,
+        Bed_Clear,
         _count,
         _last = _count - 1,
     };
