diff --git a/src/marlin_stubs/pause/M600.cpp b/src/marlin_stubs/pause/M600.cpp
index b186319d71cd3231a4014d304edd4116a593e68a..4c0c147952cd1be7de0a99e021da699eb4eccd70 100644
--- a/src/marlin_stubs/pause/M600.cpp
+++ b/src/marlin_stubs/pause/M600.cpp
@@ -117,7 +117,8 @@ void GcodeSuite::M600() {
     }
 #endif
 
-    if (do_manual_m600) {
+    if (do_manual_m600){
+        SERIAL_ECHO_MSG("M118 A1 action:pause");
         M600_manual();
     }
 
@@ -351,7 +352,7 @@ void PrusaGcodeSuite::M1601() {
         std::nullopt, std::nullopt, std::nullopt,
         std::nullopt, std::nullopt,
         pause::Settings::CalledFrom::FilamentStuck);
-
+    SERIAL_ECHO_MSG("M118 A1 action:pause");
     EMotorStallDetector::Instance().ClearReported();
 }
 #else
