diff --git a/src/gui/ScreenPrintingModel.hpp b/src/gui/ScreenPrintingModel.hpp
index 52d667d6f3dd01ff7e9eb3f926ab74696639b02c..343d8de6d00b00fbd2ff1d3c0af960012c0b8731 100644
--- a/src/gui/ScreenPrintingModel.hpp
+++ b/src/gui/ScreenPrintingModel.hpp
@@ -28,6 +28,9 @@ protected:
         Home,
         Reprint,
         Disconnect,
+        Print_OK,
+        Print_Failed,
+        Bed_Clear,
         _count,
         _last = _count - 1,
     };
@@ -44,6 +47,10 @@ protected:
         Home,
         Skip,
         Disconnect,
+        Print_OK,
+        Print_Failed,
+        Bed_Clear,
+        Disabled,
         _count,
         _last = _count - 1,
     };
