diff --git a/src/gui/MItem_menus.hpp b/src/gui/MItem_menus.hpp
index 7ac4c96bae39e8c888cbe866a2b7e3814269adb7..5da841eb8e1ec820b8aa4c50e19e13ca3bf8eb3b 100644
--- a/src/gui/MItem_menus.hpp
+++ b/src/gui/MItem_menus.hpp
@@ -170,7 +170,7 @@ protected:
 };
 
 class MI_PRUSA_CONNECT : public IWindowMenuItem {
-    static constexpr const char *const label = N_("Prusa Connect");
+    static constexpr const char *const label = N_("Simplyprint");
 
 public:
     MI_PRUSA_CONNECT();
