diff --git a/src/gui/ScreenPrintingModel.cpp b/src/gui/ScreenPrintingModel.cpp
index 47dd8eb774c168a83f7ba5e78e03e5c67d4e941d..8cbcdb2b437978acfd399fdb8d2fa40baf81e04c 100644
--- a/src/gui/ScreenPrintingModel.cpp
+++ b/src/gui/ScreenPrintingModel.cpp
@@ -23,7 +23,11 @@ constexpr const char *label_resources[] = {
     N_("Reprint"),
     N_("Home"),
     N_("Skip"),
-    N_("Disconnect")
+    N_("Disconnect"),
+    N_("Print OK"),
+    N_("Print Failed"),
+    N_("Bed Cleared"),
+    N_("Disabled"),
 };
 } // namespace
 
