diff --git a/src/gui/ScreenPrintingModel.hpp b/src/gui/ScreenPrintingModel.hpp
index 52d667d6f3dd01ff7e9eb3f926ab74696639b02c..b3ffc4819ea6d33f60c99fdb536f287f7d3a4a28 100644
--- a/src/gui/ScreenPrintingModel.hpp
+++ b/src/gui/ScreenPrintingModel.hpp
@@ -28,6 +28,9 @@ protected:
         Home,
         Reprint,
         Disconnect,
+        Print_OK,
+        Print_Failed,
+        Next_Queue_Item,
         _count,
         _last = _count - 1,
     };
@@ -44,6 +47,11 @@ protected:
         Home,
         Skip,
         Disconnect,
+        Print_OK,
+        Print_Failed,
+        Blank,
+        Next_Queue_Item,
+        Disabled,
         _count,
         _last = _count - 1,
     };
@@ -58,7 +66,11 @@ protected:
     void SetButtonLabel(BtnSocket btn, LabelRes res);
     void DisableButton(BtnSocket btn);
     void EnableButton(BtnSocket btn);
+    void SetButtonVisible(BtnSocket btn, bool visible);
+    void SetButtonRect(BtnSocket btn, Rect16 rect);
+    void ResetButtonRect(BtnSocket btn);
 
 public:
     ScreenPrintingModel(string_view_utf8 caption);
+
 };
