diff --git a/src/gui/screen_printing.hpp b/src/gui/screen_printing.hpp
index 0714477bf4858eddd62ce262edda514e40190481..291ed55230f7aeaa4676ed0f84179616b5ff56ce 100644
--- a/src/gui/screen_printing.hpp
+++ b/src/gui/screen_printing.hpp
@@ -15,6 +15,8 @@
 #include <array>
 #include <window_progress.hpp>
 #include "screen_printing_end_result.hpp"
+#include "screen_sp_bed_cleared.hpp"
+#include "screen_sp_print_result.hpp"
 
 enum class printing_state_t : uint8_t {
     INITIAL,
@@ -94,6 +96,8 @@ class screen_printing_data_t : public ScreenPrintingModel {
     bool stop_pressed;
     bool waiting_for_abort; /// flag specific for stop pressed when MBL is performed
     printing_state_t state__readonly__use_change_print_state;
+    bed_cleared state__readonly__use_change_bed_cleared_state;
+    print_succeed state__readonly__use_change_print_success_state;
 
     float last_e_axis_position;
     const Rect16 popup_rect;
@@ -105,6 +109,9 @@ class screen_printing_data_t : public ScreenPrintingModel {
     EndResultBody::DateBufferT w_etime_value_buffer;
     EndResultBody end_result_body;
 
+    SPBedClearedScreen sp_bed_cleared_body;
+    SPPrintResultScreen sp_print_result_body;
+
     enum class CurrentlyShowing {
         remaining_time, // time until end of print
         end_time, // 'date' of end of print
@@ -134,10 +141,11 @@ private:
 #endif
     void screen_printing_reprint();
     void set_pause_icon_and_label();
+    void screen_printing_start_next_queue();
+    void screen_printing_clear_bed();
     void set_tune_icon_and_label();
     void set_stop_icon_and_label();
     void change_print_state();
-
     virtual void stopAction() override;
     virtual void pauseAction() override;
     virtual void tuneAction() override;
