diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 9a0705ff3205b4a2dcf5dc3d4a043cca0351b0b2..ce6853ca9307c85f3810b1d2b4b725d9b2da8aa4 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -87,6 +87,8 @@ target_sources(
           screen_qr_error.cpp
           screen_reset_error.cpp
           screen_splash.cpp
+          screen_sp_bed_cleared.cpp
+          screen_sp_print_result.cpp
           screen_stack_overflow.cpp
           screen_sysinf.cpp
           screen_watchdog.cpp
