diff --git a/src/common/marlin_client.cpp b/src/common/marlin_client.cpp
index 019dde254..e32768c1e 100644
--- a/src/common/marlin_client.cpp
+++ b/src/common/marlin_client.cpp
@@ -18,6 +18,8 @@
 #include "utility_extensions.hpp"
 #include "tasks.hpp"
 
+#include "../gui/ScreenHandler.hpp"
+
 #if HAS_SELFTEST()
     #include <selftest_types.hpp>
     #include "printer_selftest.hpp"
@@ -385,7 +387,10 @@ bool is_print_started() {
         case State::Aborted:
         case State::Finished:
             // Went to idle - refused by GUI
-            return false;
+            // Force the GUI to close.
+            Screens::Access()->CloseAll();
+            osDelay(10);
+            return true;
         default:
             // Doing something else ‒ there's a lot of states where we are printing.
             return true;
