diff --git a/src/common/marlin_server_types/client_response.hpp b/src/common/marlin_server_types/client_response.hpp
index 169f592e41f6147bbbd4d0c7b2b641c32a53b56b..77043b20e947e4cbda1400a0b98e0f79fb0201bd 100644
--- a/src/common/marlin_server_types/client_response.hpp
+++ b/src/common/marlin_server_types/client_response.hpp
@@ -284,7 +284,7 @@ enum class PhaseNetworkSetup : PhaseUnderlyingType {
     connecting_finishable, ///< The user is connecting to a Wi-Fi. The screen offers a "Finish" button that keeps connecting on the background and "Cancel" to go back.
     connecting_nonfinishable, ///< The user is connecting to a Wi-Fi. The screen only offers a "Cancel" button to go back.
     connected,
-    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up Prusa Connect
+    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up SimplyPrint
     prusa_conect_setup, ///< Setup connect is running, waiting for it to finish
 
     no_interface_error,
