diff --git a/src/common/utils/color.hpp b/src/common/utils/color.hpp
index d93591d3a53821054b904807e53fb8e0e2b5958b..73a6c815017f8e2d83e2bea9376a66ac86027f4b 100644
--- a/src/common/utils/color.hpp
+++ b/src/common/utils/color.hpp
@@ -87,15 +87,21 @@ constexpr Color COLOR_LIGHT_GREEN = Color::from_raw(0x40b040);
 constexpr Color COLOR_PURPLE = Color::from_raw(0x800080);
 constexpr Color COLOR_TEAL = Color::from_raw(0x008080);
 constexpr Color COLOR_NAVY = Color::from_raw(0x000080);
-constexpr Color COLOR_ORANGE = Color::from_raw(0xF8651B);
+// constexpr Color COLOR_ORANGE = Color::from_raw(0xF8651B);
 constexpr Color COLOR_BRASS = Color::from_rgb(184, 134, 11); // Dark goldenrod for Signature Oak
 
+constexpr Color COLOR_PRUSA_ORANGE = Color::from_raw(0xF8651B);
+constexpr Color COLOR_SP_BLUE = Color::from_raw(0x38B6FF);
+constexpr Color COLOR_ORANGE = COLOR_SP_BLUE;
+
+constexpr Color COLOR_BRAND = COLOR_SP_BLUE;
+
 // Brand color - switches based on build variant
-#if SIGNATURE_OAK()
-                                                             // #error dead code found by automatic analyses (see BFW-5461)
-constexpr Color COLOR_BRAND = COLOR_BRASS;
-#else
-constexpr Color COLOR_BRAND = COLOR_ORANGE;
-#endif
+// #if SIGNATURE_OAK()
+//                                                              // #error dead code found by automatic analyses (see BFW-5461)
+// constexpr Color COLOR_BRAND = COLOR_BRASS;
+// #else
+// constexpr Color COLOR_BRAND = COLOR_ORANGE;
+// #endif
 
 constexpr Color COLOR_DARK_KHAKI = Color::from_raw(0xDBD76B);
