diff --git a/lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2_mk4.cpp b/lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2_mk4.cpp
index 1be081b3b4e5441932189f5c5fbeac2aee6e297d..1dc1eaa858005f34e858db6518b50e85a3b7a3ee 100644
--- a/lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2_mk4.cpp
+++ b/lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2_mk4.cpp
@@ -363,7 +363,7 @@ void MMU2::mmu_loop() {}
 
 void MMU2::CheckFINDARunout() {
     // Check for FINDA filament runout
-    if (!FindaDetectsFilament() && WhereIsFilament() == FilamentState::IN_NOZZLE) { // Check if we have filament runout detected from sensors
+    if (!FindaDetectsFilament() && WhereIsFilament() == FilamentState::AT_FSENSOR) { // Check if we have filament runout detected from sensors
         SERIAL_ECHOLNPGM("FINDA filament runout!");
         if (spool_join.get_num_joins() != 0) {
             auto join_settings = spool_join.get_spool_2(get_current_tool());
@@ -372,6 +372,7 @@ void MMU2::CheckFINDARunout() {
                 return;
             }
         }
+
         enqueue_gcode("M600");
     }
 }
