diff --git a/src/gui/resolution_480x320/screen_printing_layout.hpp b/src/gui/resolution_480x320/screen_printing_layout.hpp
index a8ab674fde99bfe599010d40e8658e0b94bbe8b3..0e970a258c4bb627a83b9ab22c7bf53a09b1d0f5 100644
--- a/src/gui/resolution_480x320/screen_printing_layout.hpp
+++ b/src/gui/resolution_480x320/screen_printing_layout.hpp
@@ -16,4 +16,7 @@ inline constexpr WindowMultiIconButton::Pngs icon_resources[] = {
     { img::reprint_80x80, img::reprint_80x80_focused, img::reprint_80x80_disabled },
     { img::disconnect_80x80, img::disconnect_80x80_focused, img::disconnect_80x80_disabled },
     { img::set_ready_80x80, img::set_ready_80x80_focused, img::set_ready_80x80_disabled },
+    { img::sp_print_ok, img::sp_print_ok_focused, img::sp_print_ok_disabled },
+    { img::sp_print_failed, img::sp_print_failed_focused, img::sp_print_failed_disabled },
+    { img::sp_next_queue_item, img::sp_next_queue_item_focused, img::sp_next_queue_item_disabled },
 };
