diff --git a/src/common/marlin_server_types/client_response.hpp b/src/common/marlin_server_types/client_response.hpp
index 0a6045fb011aa4a7165ab18074a8e3245b9173e3..4e1b66fc0b748cfe5838194ddbb63077c9c8d483 100644
--- a/src/common/marlin_server_types/client_response.hpp
+++ b/src/common/marlin_server_types/client_response.hpp
@@ -359,7 +359,7 @@ enum class PhaseNetworkSetup : PhaseUnderlyingType {
     connecting_finishable, ///< The user is connecting to a Wi-Fi. The screen offers a "Finish" button that keeps connecting on the background and "Cancel" to go back.
     connecting_nonfinishable, ///< The user is connecting to a Wi-Fi. The screen only offers a "Cancel" button to go back.
     connected,
-    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up Prusa Connect
+    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up Simplyprint
     prusa_conect_setup, ///< Setup connect is running, waiting for it to finish
 
     no_interface_error,
