diff --git a/src/gui/screen_printing.hpp b/src/gui/screen_printing.hpp
index 0714477bf4858eddd62ce262edda514e40190481..bc83f82934e15a45c01c92fbab7a349811647868 100644
--- a/src/gui/screen_printing.hpp
+++ b/src/gui/screen_printing.hpp
@@ -65,7 +65,15 @@ class screen_printing_data_t : public ScreenPrintingModel {
 
     WindowProgressCircles rotating_circles;
 #endif
+    window_icon_t sp_print_finished_icon;
+    bool was_network_print;
+    bool second_click_left;
 
+    enum class EndScreenState{
+        end_screen,
+        print_result_screen,
+    };
+    EndScreenState end_screen_state;
     window_roll_text_t w_filename;
     WindowPrintProgress w_progress;
     WindowNumbPrintProgress w_progress_txt;
@@ -105,6 +113,8 @@ class screen_printing_data_t : public ScreenPrintingModel {
     EndResultBody::DateBufferT w_etime_value_buffer;
     EndResultBody end_result_body;
 
+
+
     enum class CurrentlyShowing {
         remaining_time, // time until end of print
         end_time, // 'date' of end of print
@@ -137,7 +147,6 @@ private:
     void set_tune_icon_and_label();
     void set_stop_icon_and_label();
     void change_print_state();
-
     virtual void stopAction() override;
     virtual void pauseAction() override;
     virtual void tuneAction() override;
