diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 1975298077a885435ea09566e5a1cb221f021096..05b49c23c2a2bdb127bbcd0b590e7a280f9ae9f7 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -79,6 +79,7 @@ target_sources(
           screen_menu_temperature.cpp
           screen_menu_tune.cpp
           screen_menu_user_interface.cpp
+          screen_menu_screen_lock.cpp
           screen_messages.cpp
           screen_network_setup.cpp
           screen_print_preview.cpp
