diff --git a/src/gui/screen_network_setup.cpp b/src/gui/screen_network_setup.cpp
index 8b4f86c3626f76e060b5795bc48e508276d41b4c..c768d95dc5dd0191e6820ba2e3fa5fcd8b1ab2e7 100644
--- a/src/gui/screen_network_setup.cpp
+++ b/src/gui/screen_network_setup.cpp
@@ -463,7 +463,7 @@ class FrameAskSetupPrusaConnect : public FramePrompt {
 
 public:
     FrameAskSetupPrusaConnect(window_t *parent)
-        : FramePrompt(parent, Phase::ask_setup_prusa_connect, _("Set up Prusa Connect?"), _("Do you want to add your printer to Prusa Connect?")) {
+        : FramePrompt(parent, Phase::ask_setup_prusa_connect, _("Set up Simplyprint?"), _("Do you want to add your printer to Simplyprint?")) {
     }
 };
 
@@ -471,7 +471,7 @@ class FramePrusaConnectSetup {
 
 public:
     FramePrusaConnectSetup(window_t *) {
-        // Prusa Connect setup is a dialog for some reason, so just blockingly execute it
+        // Simplyprint setup is a dialog for some reason, so just blockingly execute it
         DialogConnectRegister::Show();
 
         // And let the FSM now that we finished afterwards
