diff --git a/Pipfile b/Pipfile
new file mode 100644
index 0000000000000000000000000000000000000000..c6921e9a7758768938c091a14450fe9bb6f159da
--- /dev/null
+++ b/Pipfile
@@ -0,0 +1,28 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+cbor2 = "~=5.7.1"
+ecdsa = "~=0.18"
+gcovr = "~=7.2"
+ndeflib = "~=0.3.3"
+numpy = "==1.26.4"
+nunavut = "~=2.3"
+pillow = "~=10.4"
+polib = "~=1.2"
+pre-commit = "*"
+pytest = "~=7.3.2"
+pyyaml = "~=6.0"
+qoi = "~=0.5"
+simple-parsing = "~=0.1.7"
+tree-sitter = "~=0.23"
+tree-sitter-cpp = "~=0.23"
+requests = "*"
+
+[dev-packages]
+
+[requires]
+python_version = "3.12"
+python_full_version = "3.12.13"
