diff --git a/src/gui/ScreenHandler.hpp b/src/gui/ScreenHandler.hpp
index f3a0eb26a2c3fefec67a77ef8f2eacee0862711b..e840ab75b46dca5bf9528f74a996a6ae36232b09 100644
--- a/src/gui/ScreenHandler.hpp
+++ b/src/gui/ScreenHandler.hpp
@@ -45,6 +45,7 @@ class Screens {
     bool close_printing = false;
 
     uint32_t timeout_tick;
+    uint32_t screen_timeout_tick;
 
     Screens(screen_node screen_creator);
     Screens(const Screens &) = delete;
@@ -84,6 +85,7 @@ public:
 
     void Draw();
     void ResetTimeout();
+    void reset_lock_timeout();
 
     void ScreenEvent(window_t *sender, GUI_event_t event, void *param);
     void WindowEvent(GUI_event_t event, void *param);
