diff --git a/src/gui/screen_network_setup.cpp b/src/gui/screen_network_setup.cpp
index 983648e9b130983344863f1595c3b68dede31ae7..b8f4727cd6a522222f20f51b98dbd31cc4509b5c 100644
--- a/src/gui/screen_network_setup.cpp
+++ b/src/gui/screen_network_setup.cpp
@@ -417,7 +417,7 @@ class FrameAskSetupPrusaConnect : public FramePrompt {
 
 public:
     FrameAskSetupPrusaConnect(window_frame_t *parent)
-        : FramePrompt(parent, Phase::ask_setup_prusa_connect, _("Set up Prusa Connect?"), _("Do you want to add your printer to Prusa Connect?")) {
+        : FramePrompt(parent, Phase::ask_setup_prusa_connect, _("Set up SimplyPrint?"), _("Do you want to add your printer to Prusa Connect?")) {
     }
 };
 
@@ -425,7 +425,7 @@ class FramePrusaConnectSetup {
 
 public:
     FramePrusaConnectSetup(window_t *) {
-        // Prusa Connect setup is a dialog for some reason, so just blockingly execute it
+        // SimplyPrint setup is a dialog for some reason, so just blockingly execute it
         DialogConnectRegister::Show();
 
         // And let the FSM now that we finished afterwards
