diff --git a/src/common/marlin_client.cpp b/src/common/marlin_client.cpp
index f4625a868bf3b2d3773acc10ce704c9daffab76c..2d8cee14c7b2244460cb1f6730a32845cad36cca 100644
--- a/src/common/marlin_client.cpp
+++ b/src/common/marlin_client.cpp
@@ -334,6 +334,7 @@ bool is_print_started() {
         case State::Aborted:
         case State::Finished:
             // Went to idle - refused by GUI
+            // TODO need to close the screen somehow?
             return false;
         default:
             // Doing something else ‒ there's a lot of states where we are printing.
