diff --git a/src/marlin_stubs/pause/M600.cpp b/src/marlin_stubs/pause/M600.cpp
index b186319d7..b40e3847d 100644
--- a/src/marlin_stubs/pause/M600.cpp
+++ b/src/marlin_stubs/pause/M600.cpp
@@ -117,7 +117,8 @@ void GcodeSuite::M600() {
     }
 #endif
 
-    if (do_manual_m600) {
+    if (do_manual_m600){
+        SERIAL_ECHO("Print Paused: Filament runout");
         M600_manual();
     }
 
@@ -351,7 +352,7 @@ void PrusaGcodeSuite::M1601() {
         std::nullopt, std::nullopt, std::nullopt,
         std::nullopt, std::nullopt,
         pause::Settings::CalledFrom::FilamentStuck);
-
+    SERIAL_ECHO("Print Paused: Filamnet Stuck");
     EMotorStallDetector::Instance().ClearReported();
 }
 #else
