diff --git a/src/connect/render.cpp b/src/connect/render.cpp
index 28738d8db..c7765ac8c 100644
--- a/src/connect/render.cpp
+++ b/src/connect/render.cpp
@@ -265,7 +265,7 @@ namespace {
             // JSON_... macros don't work in a nested switch :-(.
             if (event.type == EventType::Info) {
                 JSON_FIELD_OBJ("data");
-                    JSON_FIELD_STR("firmware", info.firmware_version) JSON_COMMA;
+                    JSON_FIELD_STR_FORMAT("firmware", "%s%s", info.firmware_version, "SP") JSON_COMMA;
                     JSON_FIELD_STR_FORMAT("printer_type", "%hhu.%hhu.%hhu", params.version.type, params.version.version, params.version.subversion) JSON_COMMA;
                     JSON_FIELD_STR("sn", info.serial_number.begin()) JSON_COMMA;
                     JSON_FIELD_BOOL("appendix", info.appendix) JSON_COMMA;
