diff --git a/src/gui/screen_printing_serial.hpp b/src/gui/screen_printing_serial.hpp
index 493e2a5db..4886d8537 100644
--- a/src/gui/screen_printing_serial.hpp
+++ b/src/gui/screen_printing_serial.hpp
@@ -6,11 +6,12 @@
 #include "status_footer.hpp"
 #include "window_text.hpp"
 #include <array>
+#include "screen_messages.hpp"
+#include "window.hpp"
+#include "window_term.hpp"
 
 class screen_printing_serial_data_t : public ScreenPrintingModel {
-    static constexpr const char *caption = N_("SERIAL PRINTING");
-
-    window_icon_t octo_icon;
+    static constexpr const char *caption = N_("SERIAL PRINTING  (MESSAGES)");
 
     int last_tick;
     enum class connection_state_t { connected,
@@ -20,8 +21,6 @@ class screen_printing_serial_data_t : public ScreenPrintingModel {
         disconnected };
     connection_state_t connection;
 
-    marlin_server::State last_state;
-
 public:
     screen_printing_serial_data_t();
 
@@ -32,4 +31,7 @@ private:
     virtual void stopAction() override;
     virtual void pauseAction() override;
     virtual void tuneAction() override;
+    window_term_t term; // Declare term object for message display
+    term_buff_t<20, 8> term_buff;
+    marlin_server::State last_state;
 };
