diff --git a/src/gui/screen_menu_connect.hpp b/src/gui/screen_menu_connect.hpp
index 8cbd1d485..585d44ba6 100644
--- a/src/gui/screen_menu_connect.hpp
+++ b/src/gui/screen_menu_connect.hpp
@@ -62,7 +62,7 @@ protected:
 };
 
 class MI_CONNECT_REGISTER : public IWindowMenuItem {
-    static constexpr const char *const label = N_("Add Printer to Connect");
+    static constexpr const char *const label = N_("Add Printer to Simplyprint");
 
 public:
     MI_CONNECT_REGISTER();
@@ -76,7 +76,7 @@ using ScreenMenuConnect__ = ScreenMenu<EFooter::Off, MI_RETURN, MI_CONNECT_ENABL
 class ScreenMenuConnect : public ScreenMenuConnect__ {
 
 public:
-    constexpr static const char *label = N_("PRUSA CONNECT");
+    constexpr static const char *label = N_("Simplyprint");
     ScreenMenuConnect();
 
 protected:
