diff --git a/src/gui/screen_network_setup.cpp b/src/gui/screen_network_setup.cpp
index 5b66b08fb..ee7395018 100644
--- a/src/gui/screen_network_setup.cpp
+++ b/src/gui/screen_network_setup.cpp
@@ -509,7 +509,7 @@ class FrameAskSetupPrusaConnect : public FrameText {
 
 public:
     FrameAskSetupPrusaConnect(window_t *parent)
-        : FrameText(parent, Phase::ask_setup_prusa_connect, _("Set up Prusa Connect?"), _("Do you want to add your printer to Prusa Connect?")) {
+        : FrameText(parent, Phase::ask_setup_prusa_connect, _("Set up Simplyprint?"), _("Do you want to add your printer to Simplyprint?")) {
     }
 };
 
@@ -517,7 +517,7 @@ class FramePrusaConnectSetup {
 
 public:
     FramePrusaConnectSetup(window_t *) {
-        // Prusa Connect setup is a dialog for some reason, so just blockingly execute it
+        // Simplyprint setup is a dialog for some reason, so just blockingly execute it
         DialogConnectRegister::Show();
 
         // And let the FSM now that we finished afterwards
