diff --git a/src/gui/screen_menu_connect.hpp b/src/gui/screen_menu_connect.hpp
index 8cbd1d485..619016b67 100644
--- a/src/gui/screen_menu_connect.hpp
+++ b/src/gui/screen_menu_connect.hpp
@@ -1,7 +1,3 @@
-/**
- * @file screen_menu_connect.hpp
- */
-
 #pragma once
 
 #include "screen_menu.hpp"
@@ -62,7 +58,7 @@ protected:
 };
 
 class MI_CONNECT_REGISTER : public IWindowMenuItem {
-    static constexpr const char *const label = N_("Add Printer to Connect");
+    static constexpr const char *const label = N_("Add Printer to Simplyprint");
 
 public:
     MI_CONNECT_REGISTER();
@@ -71,14 +67,30 @@ protected:
     virtual void click(IWindowMenu &window_menu) override;
 };
 
-using ScreenMenuConnect__ = ScreenMenu<EFooter::Off, MI_RETURN, MI_CONNECT_ENABLED, MI_CONNECT_STATUS, MI_CONNECT_ERROR, MI_CONNECT_HOST, MI_CONNECT_REGISTER, MI_CONNECT_LOAD_SETTINGS>;
+class MI_SEND_SIMPLYPRINT : public IWindowMenuItem {
+    static constexpr const char *const label = N_("Send 'Hello World' to Simplyprint");
+
+public:
+    MI_SEND_SIMPLYPRINT();
+
+protected:
+    virtual void click(IWindowMenu &window_menu) override;
+};
+
+using ScreenMenuConnect__ = ScreenMenu<EFooter::Off, MI_RETURN, MI_CONNECT_ENABLED, MI_CONNECT_STATUS, MI_CONNECT_ERROR, MI_CONNECT_HOST, MI_CONNECT_REGISTER, MI_CONNECT_LOAD_SETTINGS, MI_SEND_SIMPLYPRINT>;
 
 class ScreenMenuConnect : public ScreenMenuConnect__ {
 
 public:
-    constexpr static const char *label = N_("PRUSA CONNECT");
+    constexpr static const char *label = N_("Simplyprint");
     ScreenMenuConnect();
 
 protected:
     virtual void windowEvent(window_t *sender, GUI_event_t event, void *param) override;
+
+private:
+    bool simplyprint_task_pending = false;  // Flag to indicate if the task is pending
+
+public:
+    void setSimplyPrintTaskPending(bool pending);  // Setter to update task pending state
 };
