diff --git a/src/common/filament_sensors_handler.cpp b/src/common/filament_sensors_handler.cpp
index b1662c9e271ec833961670e54cccb3f3622eb438..da701dd3a9c3e4f8309390b8a8abdf3c46f5498a 100644
--- a/src/common/filament_sensors_handler.cpp
+++ b/src/common/filament_sensors_handler.cpp
@@ -212,7 +212,8 @@ void FilamentSensors::process_events() {
         }
 
         m600_sent = true;
-
+        SERIAL_ECHO("M118 A1 action:pause");
+        SERIAL_ECHO_MSG("M118 A1 action:pause");
         marlin_client::inject("M600 A"); // change filament
 
         log_info(FSensor, "Injected runout");
