diff --git a/src/gui/screen_printing.hpp.rej b/src/gui/screen_printing.hpp.rej
new file mode 100644
index 0000000000000000000000000000000000000000..cebfa3a54827a886deb890fa5d27aaeb172a7fdb
--- /dev/null
+++ b/src/gui/screen_printing.hpp.rej
@@ -0,0 +1,17 @@
+diff a/src/gui/screen_printing.hpp b/src/gui/screen_printing.hpp	(rejected hunks)
+@@ -61,7 +61,15 @@ class screen_printing_data_t : public ScreenPrintingModel {
+ 
+     WindowProgressCircles rotating_circles;
+ #endif
++    window_icon_t sp_print_finished_icon;
++    bool was_network_print;
++    bool second_click_left;
+ 
++    enum class EndScreenState{
++        end_screen,
++        print_result_screen,
++    };
++    EndScreenState end_screen_state;
+     window_roll_text_t w_filename;
+     WindowPrintProgress w_progress;
+     WindowNumbPrintProgress w_progress_txt;
