diff --git a/src/common/marlin_client.cpp b/src/common/marlin_client.cpp
index c56559a54c5628807e75d4b7950593d7fce43dd6..7177b2bd40df30112121029265d1d35fa3e2800b 100644
--- a/src/common/marlin_client.cpp
+++ b/src/common/marlin_client.cpp
@@ -327,6 +327,7 @@ bool is_print_started() {
         case State::Aborted:
         case State::Finished:
             // Went to idle - refused by GUI
+            // TODO need to close the screen somehow?
             return false;
         default:
             // Doing something else ‒ there's a lot of states where we are printing.
