diff --git a/src/common/marlin_server_types/client_response.hpp b/src/common/marlin_server_types/client_response.hpp
index 80acc09202566ee75b4421416be4d0b0d284e9b3..a917c8c434a27c23bf4d45430bb7ab9da603410f 100644
--- a/src/common/marlin_server_types/client_response.hpp
+++ b/src/common/marlin_server_types/client_response.hpp
@@ -377,7 +377,7 @@ enum class PhaseNetworkSetup : PhaseUnderlyingType {
     connecting_finishable, ///< The user is connecting to a Wi-Fi. The screen offers a "Finish" button that keeps connecting on the background and "Cancel" to go back.
     connecting_nonfinishable, ///< The user is connecting to a Wi-Fi. The screen only offers a "Cancel" button to go back.
     connected,
-    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up Prusa Connect
+    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up Simplyprint
     prusa_conect_setup, ///< Setup connect is running, waiting for it to finish
 
     no_interface_error,
