diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 2e465e57326948101af102bee6d6e2016fa023e2..4c5814c08f7971ecbdca70c10432f1c9fd22f2b6 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -81,6 +81,7 @@ if(BOARD IN_LIST BUDDY_BOARDS)
             w25x.cpp
             w25x_communication.cpp
             wdt.cpp
+            screen_lock.cpp
     )
   add_subdirectory(async_job)
 
