diff --git a/src/gui/MItem_menus.hpp b/src/gui/MItem_menus.hpp
index 18a6ab7631bd4251fa050b34ce48e2842d3b1523..b641b33d8c84d35503d73b72aea705c73c48e51d 100644
--- a/src/gui/MItem_menus.hpp
+++ b/src/gui/MItem_menus.hpp
@@ -88,7 +88,7 @@ using MI_MESSAGES
     = MI_SCREEN<N_("Message History"), class screen_messages_data_t>;
 
 using MI_PRUSA_CONNECT
-    = MI_SCREEN<N_("Prusa Connect"), class ScreenMenuConnect>;
+    = MI_SCREEN<N_("Simplyprint"), class ScreenMenuConnect>;
 
 using MI_PRUSALINK
     = MI_SCREEN<N_("PrusaLink"), class ScreenMenuPrusaLink>;
@@ -188,6 +188,8 @@ using MI_LEDS_SETTINGS
     = MI_SCREEN<N_("Lights Settings"), class ScreenMenuLeds>;
 #endif
 
+//TODO add code to handle setting the lock screen password
+
 class MI_SERIAL_PRINTING_SCREEN_ENABLE : public WI_ICON_SWITCH_OFF_ON_t {
     static constexpr const char *const label = N_("Serial Printing Screen");
 
