diff --git a/src/connect/planner.cpp b/src/connect/planner.cpp
index c291e1282ec01f32374c659ace04b2bef0733e5e..59fb3cd04e2a920df1a0e9310d791aa1275d1495 100644
--- a/src/connect/planner.cpp
+++ b/src/connect/planner.cpp
@@ -36,6 +36,7 @@
 #include <sys/stat.h>
 #include "stat_retry.hpp"
 #include <unistd.h>
+//#include "connect.hpp"
 
 #include <option/has_side_leds.h>
 #if HAS_SIDE_LEDS() || defined(UNITTESTS)
@@ -65,7 +66,6 @@ using Incident = ChangedPath::Incident;
 LOG_COMPONENT_REF(connect);
 
 namespace connect_client {
-
 namespace {
 
     // A note about time comparisons. We usually subtract now() and some past
@@ -709,6 +709,7 @@ JC(Resume, "No paused print to resume")
 JC(Stop, "No print to stop")
 
 void Planner::command(const Command &command, const StartPrint &params) {
+//    connect_client::is_print_network = true;
     const char *path = params.path.path();
 
     const char *reason = nullptr;
@@ -744,6 +745,7 @@ void Planner::command(const Command &command, const SendJobInfo &params) {
         command.id,
         params.job_id,
     };
+//    connect_client::is_print_network = true;
     planned_event->start_cmd_id = print_start_cmd;
 }
 
