diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index d5d7324c2cb71987a64192a36387174e1f0092cb..ec591189274f37c3b29f5b86f96ee7f8da5034b6 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -72,6 +72,7 @@ target_sources(
           screen_menu_temperature.cpp
           screen_menu_tune.cpp
           screen_menu_user_interface.cpp
+          screen_menu_screen_lock.cpp
           screen_messages.cpp
           screen_move_z.cpp
           screen_network_setup.cpp
