diff --git a/src/gui/ScreenPrintingModel.cpp b/src/gui/ScreenPrintingModel.cpp
index f45fb8e2c..8850e4b03 100644
--- a/src/gui/ScreenPrintingModel.cpp
+++ b/src/gui/ScreenPrintingModel.cpp
@@ -23,7 +23,10 @@ constexpr const char *label_resources[] = {
     N_("Reprint"),
     N_("Home"),
     N_("Skip"),
-    N_("Disconnect")
+    N_("Disconnect"),
+    N_("Bed Cleared"),
+    N_("Print Good"),
+    N_("Print Failed"),
 };
 } // namespace
 
