diff --git a/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py b/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py
index 2f67798cc805a7ae80445da9461751c3507c7a8f..61f2a1f6ded380e44521f8d74fd99471e39ef29e 100644
--- a/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py
+++ b/lib/Prusa-Error-Codes/prusaerrors/shared/codes.py
@@ -51,7 +51,7 @@ class Category(IntEnum):
     MECHANICAL = 1  # Mechanical failures, engines XYZ, tower
     TEMPERATURE = 2  # Temperature measurement, thermistors, heating
     ELECTRICAL = 3  # Electrical, MINDA, FINDA, Motion Controller, …
-    CONNECTIVITY = 4  # Connectivity - Wi - Fi, LAN, Prusa Connect Cloud
+    CONNECTIVITY = 4  # Connectivity - Wi - Fi, LAN, Simplyprint Cloud
     SYSTEM = 5  # System - BSOD, ...
     BOOTLOADER = 6  #
     WARNINGS = 7  # Category-less warnings
