diff --git a/src/common/utils/color.hpp b/src/common/utils/color.hpp
index f581c0dcf0bba82ffebdd2f957e42422335d2344..8a7e17f361d62bf5b4160f078cf3ed26f9666ccb 100644
--- a/src/common/utils/color.hpp
+++ b/src/common/utils/color.hpp
@@ -84,5 +84,8 @@ constexpr Color COLOR_LIGHT_GREEN = Color::from_raw(0x40b040);
 constexpr Color COLOR_PURPLE = Color::from_raw(0x800080);
 constexpr Color COLOR_TEAL = Color::from_raw(0x008080);
 constexpr Color COLOR_NAVY = Color::from_raw(0x000080);
-constexpr Color COLOR_ORANGE = Color::from_raw(0xF8651B);
+//constexpr Color COLOR_ORANGE = Color::from_raw(0xF8651B);
 constexpr Color COLOR_DARK_KHAKI = Color::from_raw(0xDBD76B);
+constexpr Color COLOR_PRUSA_ORANGE = Color::from_raw(0xF8651B);
+constexpr Color COLOR_SP_BLUE = Color::from_raw(0x38B6FF);
+constexpr Color COLOR_ORANGE = COLOR_SP_BLUE;
