diff --git a/src/guiapi/src/window_msgbox.cpp b/src/guiapi/src/window_msgbox.cpp
index 580baae41575302cfdcde5313ea07c4943b464ad..5c134e3b3fc772416268e76827400df421203f8a 100644
--- a/src/guiapi/src/window_msgbox.cpp
+++ b/src/guiapi/src/window_msgbox.cpp
@@ -199,9 +199,9 @@ MsgBoxIconnedError::MsgBoxIconnedError(Rect16 rect, const PhaseResponses &resp,
     text.SetAlignment(Align_t::LeftCenter());
     icon.SetRect(getIconRect());
     AdjustLayout(text, icon);
-    SetBackColor(COLOR_ORANGE);
-    text.SetBackColor(COLOR_ORANGE);
-    icon.SetBackColor(COLOR_ORANGE);
+    SetBackColor(COLOR_PRUSA_ORANGE);
+    text.SetBackColor(COLOR_PRUSA_ORANGE);
+    icon.SetBackColor(COLOR_PRUSA_ORANGE);
 
     if (!pButtons) { // pButtons can never be null
         assert("unassigned msgbox");
