diff --git a/src/common/marlin_client.cpp b/src/common/marlin_client.cpp
index 3f3ae82478da65254d4aba60c22072263e771a55..e97d83d7c7756517abba5fa34eda85bb0f1c6ed8 100644
--- a/src/common/marlin_client.cpp
+++ b/src/common/marlin_client.cpp
@@ -326,6 +326,7 @@ bool is_print_started() {
         case State::Aborted:
         case State::Finished:
             // Went to idle - refused by GUI
+            // TODO need to close the screen somehow?
             return false;
         default:
             // Doing something else ‒ there's a lot of states where we are printing.
