diff --git a/src/connect/planner.cpp b/src/connect/planner.cpp
index f4c01b983ef232ea3b102ce07216237b2681e101..d708795e0cdea50b4e68234d29ca165bf864b957 100644
--- a/src/connect/planner.cpp
+++ b/src/connect/planner.cpp
@@ -2,6 +2,9 @@
 #include "netdev.h"
 #include "printer.hpp"
 
+#include <common/data_exchange.hpp>
+#include <common/sys.hpp>
+
 #include <path_utils.h>
 #include <filename_type.hpp>
 #include <logging/log.hpp>
@@ -65,7 +68,6 @@ using Incident = ChangedPath::Incident;
 LOG_COMPONENT_REF(connect);
 
 namespace connect_client {
-
 namespace {
 
     // A note about time comparisons. We usually subtract now() and some past
@@ -717,6 +719,7 @@ JC(Resume, "No paused print to resume")
 JC(Stop, "No print to stop")
 
 void Planner::command(const Command &command, const StartPrint &params) {
+//    connect_client::is_print_network = true;
     const char *path = params.path.path();
 
     const char *reason = nullptr;
@@ -752,6 +755,7 @@ void Planner::command(const Command &command, const SendJobInfo &params) {
         command.id,
         params.job_id,
     };
+//    connect_client::is_print_network = true;
     planned_event->start_cmd_id = print_start_cmd;
 }
 
