diff --git a/src/connect/planner.cpp b/src/connect/planner.cpp
index f4c01b983ef232ea3b102ce07216237b2681e101..a4413424150d7f4c78301fd5e0626a932c277641 100644
--- a/src/connect/planner.cpp
+++ b/src/connect/planner.cpp
@@ -36,6 +36,7 @@
 #include <sys/stat.h>
 #include "stat_retry.hpp"
 #include <unistd.h>
+//#include "connect.hpp"
 
 #include <option/has_side_leds.h>
 #if HAS_SIDE_LEDS() || defined(UNITTESTS)
@@ -65,7 +66,6 @@ using Incident = ChangedPath::Incident;
 LOG_COMPONENT_REF(connect);
 
 namespace connect_client {
-
 namespace {
 
     // A note about time comparisons. We usually subtract now() and some past
@@ -717,6 +717,7 @@ JC(Resume, "No paused print to resume")
 JC(Stop, "No print to stop")
 
 void Planner::command(const Command &command, const StartPrint &params) {
+//    connect_client::is_print_network = true;
     const char *path = params.path.path();
 
     const char *reason = nullptr;
@@ -752,6 +753,7 @@ void Planner::command(const Command &command, const SendJobInfo &params) {
         command.id,
         params.job_id,
     };
+//    connect_client::is_print_network = true;
     planned_event->start_cmd_id = print_start_cmd;
 }
 
