diff --git a/src/connect/planner.hpp b/src/connect/planner.hpp
index 7a2b6af545bc7962ba00291deaa87cb2e210e4ab..058c42caea396a6e586552548c737ba5cc0b426c 100644
--- a/src/connect/planner.hpp
+++ b/src/connect/planner.hpp
@@ -192,6 +192,7 @@ private:
     void command(const Command &, const SetValue &);
     void command(const Command &, const CancelObject &);
     void command(const Command &, const UncancelObject &);
+    void command(const Command &, const InitiateFirmwareUpdate &);
 
     void handle_transfer_result(const Command &command, transfers::Transfer::BeginResult result);
     void handle_cancel_object_command(const Command &command, uint16_t object_id, bool set_cancelled);
