diff --git a/src/connect/command.cpp b/src/connect/command.cpp
index fed486d0941c094fdf9bb571fa462e7cb6e330db..74b385ca0764f56ecc7d76c441c56e2ec3652a8c 100644
--- a/src/connect/command.cpp
+++ b/src/connect/command.cpp
@@ -190,7 +190,8 @@ Command Command::parse_json_command(CommandId id, char *body, size_t body_size,
             // which command will be coming from which Connect instance ->
             // accept both, handle the same way.
             T("START_CONNECT_DOWNLOAD", StartInlineDownload, ARGS_INLINE_DOWN)
-            T("START_ENCRYPTED_DOWNLOAD", StartEncryptedDownload, ARGS_ENC_DOWN) { // else is part of the previous T
+            T("START_ENCRYPTED_DOWNLOAD", StartEncryptedDownload, ARGS_ENC_DOWN)
+            T("INITIATE_FIRMWARE_UPDATE", InitiateFirmwareUpdate, NO_ARGS) { // else is part of the previous T
                 return;
             }
         }
