diff --git a/src/connect/command.hpp b/src/connect/command.hpp
index 4cba42a1711e29a6760aeb4084aee17db19bad9d..66f241fe12de2f8224e3a79b31a91413bc2ef5dd 100644
--- a/src/connect/command.hpp
+++ b/src/connect/command.hpp
@@ -135,8 +135,9 @@ struct CancelObject {
 struct UncancelObject {
     uint16_t id;
 };
+struct InitiateFirmwareUpdate {};
 
-using CommandData = std::variant<UnknownCommand, BrokenCommand, GcodeTooLarge, ProcessingOtherCommand, ProcessingThisCommand, Gcode, SendInfo, SendJobInfo, SendFileInfo, SendTransferInfo, PausePrint, ResumePrint, StopPrint, StartPrint, SetPrinterReady, CancelPrinterReady, SetPrinterIdle, StartEncryptedDownload, StartInlineDownload, DeleteFile, DeleteFolder, CreateFolder, StopTransfer, SetToken, ResetPrinter, SendStateInfo, DialogAction, SetValue, CancelObject, UncancelObject>;
+using CommandData = std::variant<UnknownCommand, BrokenCommand, GcodeTooLarge, ProcessingOtherCommand, ProcessingThisCommand, Gcode, SendInfo, SendJobInfo, SendFileInfo, SendTransferInfo, PausePrint, ResumePrint, StopPrint, StartPrint, SetPrinterReady, CancelPrinterReady, SetPrinterIdle, StartEncryptedDownload, StartInlineDownload, DeleteFile, DeleteFolder, CreateFolder, StopTransfer, SetToken, ResetPrinter, SendStateInfo, DialogAction, SetValue, CancelObject, UncancelObject, InitiateFirmwareUpdate>;
 
 struct Command {
     CommandId id;
