diff --git a/src/common/marlin_client.cpp b/src/common/marlin_client.cpp
index 387e575acab106ffd4d591ccf93c70589429e9fe..ed36c8d6824be059b1d065abc22b2415afa1aa02 100644
--- a/src/common/marlin_client.cpp
+++ b/src/common/marlin_client.cpp
@@ -351,6 +351,7 @@ bool is_print_started() {
         case State::Aborted:
         case State::Finished:
             // Went to idle - refused by GUI
+            // TODO need to close the screen somehow?
             return false;
         default:
             // Doing something else ‒ there's a lot of states where we are printing.
