diff --git a/src/gui/screen_printing.hpp b/src/gui/screen_printing.hpp
index 1e438067277fd1f4fdc41d47bfb82ef4493b89a4..449dc3b1ebef967aa6961656b95f353101104b22 100644
--- a/src/gui/screen_printing.hpp
+++ b/src/gui/screen_printing.hpp
@@ -103,6 +103,16 @@ class screen_printing_data_t : public ScreenPrintingModel {
     EndResultBody::DateBufferT w_etime_value_buffer;
     EndResultBody end_result_body;
 
+    window_icon_t sp_print_finished_icon;
+    bool was_network_print;
+    bool second_click_left;
+
+    enum class EndScreenState{
+        end_screen,
+        print_result_screen,
+    };
+    EndScreenState end_screen_state;
+
     enum class CurrentlyShowing {
         remaining_time, // time until end of print
         end_time, // 'date' of end of print
@@ -139,7 +149,6 @@ private:
     void set_tune_icon_and_label();
     void set_stop_icon_and_label();
     void change_print_state();
-
     virtual void stopAction() override;
     virtual void pauseAction() override;
     virtual void tuneAction() override;
