diff --git a/lib/Marlin/Marlin/src/module/temperature.cpp b/lib/Marlin/Marlin/src/module/temperature.cpp
index 0fcd0d88b16d5dd9449737e4b477c5931d0bb763..a384d9efd37c06b5e513694bc632c9e984b3acc3 100644
--- a/lib/Marlin/Marlin/src/module/temperature.cpp
+++ b/lib/Marlin/Marlin/src/module/temperature.cpp
@@ -3892,7 +3892,11 @@ void Temperature::isr() {
       SERIAL_ECHOPAIR(" C@:", getHeaterPower(H_CHAMBER));
     #elif HAS_CHAMBER_API()
       auto current_chamber_temperature = buddy::chamber().current_temperature();
-      if (current_chamber_temperature.has_value()) SERIAL_ECHOPAIR(" C@:", current_chamber_temperature.value());
+      auto target_chamber_temperature = buddy::chamber().target_temperature();
+      if (current_chamber_temperature.has_value()){
+          SERIAL_ECHOPAIR(" C:", current_chamber_temperature.value());
+          SERIAL_ECHOPAIR("/", target_chamber_temperature.has_value() ? target_chamber_temperature.value() : 0.0f);
+      }
     #endif
 
     #if HAS_TEMP_HEATBREAK
