diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index ac4fd01d8db9afc8c25b80468287e3e4b1fd82dd..7d2b71b3fc1abdee237229813ee0bab070b9bdb0 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -80,6 +80,7 @@ target_sources(
           screen_menu_temperature.cpp
           screen_menu_tune.cpp
           screen_menu_user_interface.cpp
+          screen_menu_screen_lock.cpp
           screen_messages.cpp
           screen_move_z.cpp
           screen_network_setup.cpp
