diff --git a/src/gui/ScreenPrintingModel.hpp b/src/gui/ScreenPrintingModel.hpp
index 1204ada026338d1fe291e03d7b6ca94ed8715fc3..be2fd0ae048889b572078323198667c336b6c774 100644
--- a/src/gui/ScreenPrintingModel.hpp
+++ b/src/gui/ScreenPrintingModel.hpp
@@ -29,6 +29,11 @@ protected:
         Reprint,
         Disconnect,
         SetReady,
+#if HAS_LARGE_DISPLAY()
+        Print_OK,
+        Print_Failed,
+        Next_Queue_Item,
+#endif
         _count,
         _last = _count - 1,
     };
@@ -46,6 +51,11 @@ protected:
         Skip,
         Disconnect,
         SetReady,
+        Print_OK,
+        Print_Failed,
+        Blank,
+        Next_Queue_Item,
+        Disabled,
         _count,
         _last = _count - 1,
     };
@@ -60,6 +70,11 @@ protected:
     void SetButtonLabel(BtnSocket btn, LabelRes res);
     void DisableButton(BtnSocket btn);
     void EnableButton(BtnSocket btn);
+    void SetButtonVisible(BtnSocket btn, bool visible);
+    void SetButtonRect(BtnSocket btn, Rect16 rect);
+    void ResetButtonRect(BtnSocket btn);
+    void SetLabelVisible(BtnSocket idx, bool visible);
+
 
 public:
     ScreenPrintingModel(const string_view_utf8 &caption);
