diff --git a/src/mmu2/mmu2_error_converter.cpp b/src/mmu2/mmu2_error_converter.cpp
index 88e94d0761902826432540638d4127b1ae4c5a2d..5156235f486ac2bae2b49a15fa216cd6a2ffe274 100644
--- a/src/mmu2/mmu2_error_converter.cpp
+++ b/src/mmu2/mmu2_error_converter.cpp
@@ -4,6 +4,7 @@
 #include "../../lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2_error_converter.h"
 #include "../../lib/Prusa-Firmware-MMU/src/logic/error_codes.h"
 #include "utility_extensions.hpp"
+#include "../../lib/Marlin/Marlin/src/core/serial.h"
 
 #include <algorithm>
 
