diff --git a/src/gui/MItem_menus.hpp b/src/gui/MItem_menus.hpp
index 471271a6bd477ddb3069b86163cefea8926a2403..8e078d9dc5d838582d3f6b65127f50946f9d6f91 100644
--- a/src/gui/MItem_menus.hpp
+++ b/src/gui/MItem_menus.hpp
@@ -103,7 +103,7 @@ using MI_MESSAGES
     = MI_SCREEN<N_("Message History"), class screen_messages_data_t>;
 
 using MI_PRUSA_CONNECT
-    = MI_SCREEN<N_("Prusa Connect"), class ScreenMenuConnect>;
+    = MI_SCREEN<N_("SimplyPrint"), class ScreenMenuConnect>;
 
 using MI_PRUSALINK
     = MI_SCREEN<N_("PrusaLink"), class ScreenMenuPrusaLink>;
