diff --git a/src/gui/dialogs/DialogConnectReg.cpp b/src/gui/dialogs/DialogConnectReg.cpp
index 8646499ad7bd481df53ac052c16c711ae7356866..e1c505942be61eafd8b439b9bf990b8463665488 100644
--- a/src/gui/dialogs/DialogConnectReg.cpp
+++ b/src/gui/dialogs/DialogConnectReg.cpp
@@ -38,11 +38,11 @@ DialogConnectRegister::DialogConnectRegister()
     text_detail.SetAlignment(Align_t::Center());
     line.SetBackColor(COLOR_ORANGE);
 
-    title.SetText(_("Prusa Connect - printer setup"));
+    title.SetText(_("Simplyprint - printer setup"));
 
     last_seen_status = std::make_tuple(connect_client::ConnectionStatus::Unknown, connect_client::OnlineError::NoError, std::nullopt);
 
-    text_state.SetText(_("Registering the printer to Prusa Connect..."));
+    text_state.SetText(_("Registering the printer to Simplyprint..."));
     text_attempt.Hide();
 
     // Show these only after we get the code.
@@ -82,7 +82,7 @@ void DialogConnectRegister::windowEvent(window_t *sender, GUI_event_t event, voi
         switch (get<0>(last_seen_status)) {
         case ConnectionStatus::RegistrationRequesting:
         case ConnectionStatus::RegistrationCode:
-            close = MsgBoxWarning(_("Prusa Connect setup is not finished. Do you want to exit and abort the process?"), Responses_YesNo)
+            close = MsgBoxWarning(_("Simplyprint setup is not finished. Do you want to exit and abort the process?"), Responses_YesNo)
                 == Response::Yes;
             break;
         default:
@@ -111,7 +111,7 @@ void DialogConnectRegister::windowEvent(window_t *sender, GUI_event_t event, voi
                 // therefore the wizard is of little use to them.
 
                 qr_registration_code.get_string_builder()
-                    .append_string("https://connect.prusa3d.com/add/")
+                    .append_string("https://simplyprint.io/add/")
                     .append_string(code);
                 showQR(qr_registration_code);
 
@@ -126,14 +126,14 @@ void DialogConnectRegister::windowEvent(window_t *sender, GUI_event_t event, voi
 #if HAS_MINI_DISPLAY()
                 text_state.SetText(_("Scan the QR code using the Prusa app or camera, or visit prusa.io/add"));
 #else
-                text_state.SetText(_("1. Scan the QR code using the Prusa app or camera, or visit prusa.io/add\n\n2. Log in"));
+                text_state.SetText(_("1. Scan the QR code using the Simplyprint app or camera, or visit simplyprint.io/\n\n2. Log in"));
 #endif
                 break;
                 ;
             }
             case ConnectionStatus::RegistrationDone: {
                 hideDetails();
-                text_state.SetText(_("Registration successful, continue at connect.prusa3d.com"));
+                text_state.SetText(_("Registration successful, continue at simplyprint.io"));
                 connect_client::leave_registration();
                 left_registration = true;
                 break;
@@ -181,7 +181,7 @@ void DialogConnectRegister::windowEvent(window_t *sender, GUI_event_t event, voi
                 showQR(qr_error);
 
                 _(moreDetailTxt).copyToRAM(error_detail_buffer, sizeof(error_detail_buffer)); // Translation
-                snprintf(detail_buffer, sizeof(detail_buffer), "%s:\nprusa.io/%d", error_detail_buffer, static_cast<int>(error.err_code));
+                snprintf(detail_buffer, sizeof(detail_buffer), "%s:\nsimplyprint.io/%d", error_detail_buffer, static_cast<int>(error.err_code));
                 text_detail.SetText(string_view_utf8::MakeRAM((const uint8_t *)detail_buffer));
                 text_detail.Invalidate();
                 break;
