diff --git a/src/common/marlin_server_types/client_response.hpp b/src/common/marlin_server_types/client_response.hpp
index 0e35c9ac1b2e8d5765a742e5030f00cb62c8e641..89f3f06ef287a9d9687d79bff5be9fdc5924581e 100644
--- a/src/common/marlin_server_types/client_response.hpp
+++ b/src/common/marlin_server_types/client_response.hpp
@@ -356,7 +356,7 @@ enum class PhaseNetworkSetup : PhaseUnderlyingType {
     connecting_finishable, ///< The user is connecting to a Wi-Fi. The screen offers a "Finish" button that keeps connecting on the background and "Cancel" to go back.
     connecting_nonfinishable, ///< The user is connecting to a Wi-Fi. The screen only offers a "Cancel" button to go back.
     connected,
-    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up Prusa Connect
+    ask_setup_prusa_connect, ///< Prompts the user if he wants to set up Simplyprint
     prusa_conect_setup, ///< Setup connect is running, waiting for it to finish
 
     no_interface_error,
